### ISSUE TITLE
Watchman killing jenkins slave gear during build

### ISSUE THREAD

**Issue by caruccio** [openshift/origin-server#6190]
Watchman will restart jenkins slave gear during building.

Steps to reproduce:
---
- Create a jenkins server
- Create a test app (any lang) and add jenkins-client to it
- Create a `.openshift/action_hooks/build` file with content:

    echo sleeping...
    sleep 120
    echo ...wake up

- Publish test app

Expected result
---
The build should  succeed

Received result
---
The build fails

How to verify
---
SSH into node as root and `tail /var/log/messages`. The following messages should appear:
```
Jul  6 14:31:17 node4 watchman[2408]: watchman deleted stop lock for gear 559a90d299fc77f34f00028f because the state of the gear was building
Jul  6 14:36:31 node4 watchman[2408]: watchman restarted user 559a90d299fc77f34f00028f: application pythonbldr (retries: 0)
```

Possible solution
---
Handle `State::BUILDING` [here](https://github.com/openshift/origin-server/blob/6ffa67d0bcac6cb011e6bedcc5980a70844ae2f4/node-util/conf/watchman/plugins.d/gear_state_plugin.rb#L86). Now it falls [here](https://github.com/openshift/origin-server/blob/6ffa67d0bcac6cb011e6bedcc5980a70844ae2f4/node-util/conf/watchman/plugins.d/gear_state_plugin.rb#L108)

**Comment by tiwillia** [comment]
@caruccio, are you using the latest codebase? I am unable to reproduce the issue in my environment using the steps you provided.

This issue is documented in [BZ #1134686](https://bugzilla.redhat.com/show_bug.cgi?id=1134686) and should have been fixed already with cc0961110326ed25ab13691b20ed4a6a88a295ab

**Comment by caruccio** [comment]
Sorry, forgot to mention i'm running release 4.
Looks like cc0961110326ed25ab13691b20ed4a6a88a295ab will do the job. Thanks!
